frontend/src/components/Forms/MineRecordForm.js

```diff
@@ -65,39 +117,134 @@ export class MineRecordForm extends Component {
+  addField = (event, fields) => {
+    const totalTypes = this.props.currentMineTypes
+      ? fields.length + this.props.currentMineTypes.length
+      : fields.length;
+    event.preventDefault();
+    if (totalTypes === 4) {
+      notification.error({
+        message: "You cannot have more than 4 tenures associated with a mine",
+        duration: 10,
+      });
+    } else {
+      fields.push({});
+      this.setActiveKey([fields.length.toString()]);
+    }
+  };
+
+  handleTenureCodeUpdate = (opt) => {
+    this.setState((prevState) => ({
+      usedTenureTypes: [...prevState.usedTenureTypes, opt],
+    }));
   };
 
   createPanelHeader = (index, fields) => (
     <div className="inline-flex between">
-      <Form.Item style={{ marginTop: "15px" }} label={`Mine Type ${index + 1}`} />
-      <Button
-        ghost
-        onClick={(event) => {
-          this.removeField(event, fields, index);
-        }}
-      >
-        <Icon type="minus-circle" theme="outlined" style={{ color: "#BC2929" }} />
-      </Button>
+      <Form.Item style={{ marginTop: "15px" }} label={`New Mine Type: ${index + 1}`} />
+      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
+      <div onClick={(event) => event.stopPropagation()}>
+        <Popconfirm
+          placement="topRight"
+          title={`Are you sure you want to remove Mine Type ${index + 1}?`}
+          onConfirm={(event) => {
+            this.removeField(event, fields, index);
+          }}
+          okText="Yes"
+          cancelText="No"
+        >
+          <Button ghost>
+            <Icon type="minus-circle" theme="outlined" style={{ color: "#BC2929" }} />
+          </Button>
+        </Popconfirm>
+      </div>
+    </div>
+  );
+
+  createExistingPanelHeader = (mineTenureCode) => (
+    <div className="inline-flex between">
+      <Form.Item
+        style={{ marginTop: "15px" }}
+        label={`Existing Mine Type: ${this.props.mineTenureHash[mineTenureCode]}`}
+      />
+      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
+      <div onClick={(event) => event.stopPropagation()}>
+        <Popconfirm
+          placement="topRight"
+          title={`Are you sure you want to remove Mine Type ${
+            this.props.mineTenureHash[mineTenureCode]
+          }?`}
+          onConfirm={(event) => this.props.handleDelete(event, mineTenureCode)}
+          okText="Yes"
+          cancelText="No"
+        >
+          <Button ghost>
+            <Icon type="minus-circle" theme="outlined" style={{ color: "#BC2929" }} />
+          </Button>
+        </Popconfirm>
+      </div>
     </div>
   );
 
   render() {
     const renderTypeSelect = ({ fields }) => (
       <div>
         <Collapse activeKey={this.state.activeKey} onChange={this.setActiveKey}>
+          {this.props.currentMineTypes &&
+            this.props.currentMineTypes.map((type) => (
```

Comment:
At a quick scan, it doesn't look like you reference `type` on its own after this. Could likely just destructure out `mine_tenure_type_code`

Reply:
`mine_tenure_type_code, mine_disturbance_code, and mine_commodity_code` are all used,

is it preferred to destructure them all out?

Reply:
Your call. If it makes the code harder to read then possibly not